extensions/sql-migration/src/dialog/createResourceGroup/createResourceGroupDialog.ts

```diff
@@ -0,0 +1,185 @@
+
+import * as azdata from 'azdata';
+import { azureResource } from 'azureResource';
+import { EventEmitter } from 'events';
+import { createResourceGroup } from '../../api/azure';
+import * as constants from '../../constants/strings';
+
+export class CreateResourceGroupDialog {
+	private _dialogObject!: azdata.window.Dialog;
+	private _view!: azdata.ModelView;
+	private _creationEvent: EventEmitter = new EventEmitter;
+
+	constructor(private _azureAccount: azdata.Account, private _subscription: azureResource.AzureResourceSubscription, private _location: string) {
+		this._dialogObject = azdata.window.createModelViewDialog(
+			'',
+			'CreateResourceGroupDialog',
+			550,
+			'callout',
+			'below',
+			false,
+			true,
+			<azdata.window.IDialogProperties>{
+				height: 20,
+				width: 20,
+				xPos: 0,
+				yPos: 0
+			}
+		);
+	}
+
+	async initialize(): Promise<azureResource.AzureResourceResourceGroup> {
+		let tab = azdata.window.createTab('');
+		await tab.registerContent(async (view: azdata.ModelView) => {
+			this._view = view;
+
+			const resourceGroupDescription = view.modelBuilder.text().withProps({
+				value: constants.RESOURCE_GROUP_DESCRIPTION,
+				CSSStyles: {
+					'font-size': '13px',
+					'margin-bottom': '10px'
+				}
+			}).component();
+			const nameLabel = view.modelBuilder.text().withProps({
+				value: constants.NAME,
+				CSSStyles: {
+					'font-size': '13px',
+					'font-weight': 'bold',
+				}
+			}).component();
+
+			const resoruceGroupName = view.modelBuilder.inputBox().withValidation(c => {
+				let valid = false;
+				if (c.value!.length > 0 && c.value!.length <= 90 && /^[-\w\._\(\)]+$/.test(c.value!)) {
+					valid = true;
+				}
+				okButton.enabled = valid;
+				return valid;
+			}).component();
+
+			resoruceGroupName.onTextChanged(e => {
+				errorBox.updateCssStyles({
+					'display': 'none'
+				});
+			});
+
+			const okButton = view.modelBuilder.button().withProps({
+				label: constants.OK,
+				width: '80px',
+				enabled: false
+			}).component();
+
+			okButton.onDidClick(async e => {
+				errorBox.updateCssStyles({
+					'display': 'none'
+				});
+				okButton.enabled = false;
+				cancelButton.enabled = false;
+				loading.loading = true;
+				try {
+					const resourceGroup = await createResourceGroup(this._azureAccount, this._subscription, resoruceGroupName.value!, this._location);
+					this._creationEvent.emit('done', resourceGroup);
+				} catch (e) {
+					errorBox.updateCssStyles({
+						'display': 'inline'
+					});
+					errorBox.text = e.toString();
+					cancelButton.enabled = true;
+					resoruceGroupName.validate();
+				} finally {
+					loading.loading = false;
+				}
+			});
+
+			const cancelButton = view.modelBuilder.button().withProps({
+				label: constants.CANCEL,
+				width: '80px'
+			}).component();
+
+			cancelButton.onDidClick(e => {
+				this._creationEvent.emit('done', undefined);
+			});
+
+			const loading = view.modelBuilder.loadingComponent().withProps({
```

Comment:
Would suggest adding a more specific `loadingText` and `loadingCompletedText` instead of the default